### PR TITLE
docs: improve webpack migration

### PR DESCRIPTION
**Pull request by SyMind**
## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: website/docs/en/guide/migration/webpack.mdx [hunk 1]
```diff
@@ -1,110 +1,169 @@
...
-## Configuration File
+:::tip
+Rspack commands can specify the configuration file with `-c` or `--config` like webpack commands.
+Unlike webpack, if you do not explicitly specify a configuration file, Rspack defaults to using `rspack.config.js`.
+:::
 
-Please rename the `webpack.config.js` file to `rspack.config.js`.
+## Updating configuration
 
-Rspack offers configurations similar to webpack. You can refer to [Configure Rspack](/config/index) to see the configurations supported by Rspack.
```
Comment (chenjiahan) [inline website/docs/en/guide/migration/webpack.mdx:59]: We should preserve the `[Configure Rspack](/config/index)` link

File: website/docs/en/guide/migration/webpack.mdx [hunk 2]
```diff
@@ -1,110 +1,169 @@
...
 ```diff title="package.json"
 {
   "scripts": {
--   "serve": "webpack serve -c webpack.config.js",
--   "build": "webpack build -c webpack.config.js",
-+   "serve": "rspack serve -c rspack.config.js",
-+   "build": "rspack build -c rspack.config.js",
+-   "serve": "webpack serve",
+-   "build": "webpack build",
++   "serve": "rspack serve -c webpack.config.js",
```
[the remark this answers is not in the document]
Comment (SyMind) [reply website/docs/en/guide/migration/webpack.mdx:33]: Retaining the `webpack.config.js` filename allows changes to be tracked using git diff.

File: website/docs/en/guide/migration/webpack.mdx [hunk 3]
```diff
@@ -1,110 +1,169 @@
...
+- const webpack = require('webpack');
++ const rspack = require('@rspack/core');
 
 module.exports = {
-  plugins: [isDev && new webpack.HotModuleReplacementPlugin()].filter(Boolean),
-};
+  //...
+  plugins: [
+-   new webpack.HotModuleReplacementPlugin({
++   new rspack.HotModuleReplacementPlugin({
```
[the remark this answers is not in the document]
Comment (SyMind) [reply website/docs/en/guide/migration/webpack.mdx:83]: I want to make the example more concise and avoid unnecessary noise. Maybe i can change another plugin.

File: website/docs/en/guide/migration/webpack.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (chenjiahan) [reply website/docs/en/guide/migration/webpack.mdx:33]: Git can identify file renames by content similarity, we do not need to consider this

File: website/docs/en/guide/migration/webpack.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (chenjiahan) [reply website/docs/en/guide/migration/webpack.mdx:83]: Yeah another plugin can be better

File: website/docs/en/guide/migration/webpack.mdx [hunk 4]
```diff
@@ -1,110 +1,169 @@
...
 
-For projects not using webpack v5, there are other migration guides that can be referenced:
+This document is primarily aimed at projects using webpack 5. Since Rspack's API and configuration align with webpack 5.
+For projects not using webpack 5, there are other migration guides that can be referenced:
 
 - For projects using webpack v4 or earlier versions, you can refer to [webpack - To v5 from v4](https://webpack.js.org/migrate/5/) to understand the differences.
 - For projects using create-react-app or CRACO, you can refer to [Migrating Create React App](/guide/migration/cra).
 - For projects using Vue CLI, you can refer to [Rsbuild - Migrating from Vue CLI](https://rsbuild.dev/guide/migration/vue-cli).
 
-## Notes
```
[the remark this answers is not in the document]
Comment (SyMind) [reply website/docs/en/guide/migration/webpack.mdx:13]: This content does not directly help with the migration; it is either unnecessary or too general.

1. The minimum node version supported by Rspack should be documented consistently elsewhere. Mentioning it here only adds to the maintenance burden.
2. The differences you will encounter between Rspack and Webpack during migration are already detailed in the sections below.
3. The best practices for achieving optimal performance with Rspack are outlined in the loader migration section.

File: website/docs/en/guide/migration/webpack.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (SyMind) [reply website/docs/en/guide/migration/webpack.mdx:33]: But I write a tip below that Rspack default used  `rspack.config.js`.

[the remark this answers is not in the document]
Comment (chenjiahan) [reply website/docs/en/guide/migration/webpack.mdx:33]: How about just removing `-c webpack.config.js` from the example, this is an anti-pattern.

**Review by chenjiahan — DISMISSED** [review]
Awesome 👍

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**